Writing an introduction for the Contribution guide (#161)
rust-unofficial/patterns: CONTRIBUTING.md

# Contributing

## Introduction

> This book is a catalogue of Rust programming techniques, (anti-)patterns, and idioms and other explanations.

Reviewer: ```suggestion
This book is a catalogue of Rust programming techniques, (anti-)patterns, idioms and other explanations.
```

I suggest having just one `and` here, by replacing the `, and idioms` with just `, idioms`.

I'm referencing this: https://english.stackexchange.com/a/187872, which states the following:

`
There is nothing that would stop you from using 'and' multiple times, but if you are to list things, using a comma is better. 
Example: "I like to play football, ride my bike, play guitar and hang with friends." 
Instead of "I like to play football and ride my bike and play guitar and hang with friends".
`

> It is a compilation of collective (sometimes implicit) knowledge or experience that was created with collaborative effort.

Reviewer: I'm not really sure how to phrase this, but maybe someone else has an idea?

Author: ```suggestion
It is a compilation of collective (sometimes implicit) knowledge as well as experiences that have emerged through collaborative work.
```

> These contained information are __no rules__, they should be understood as guidelines.

Reviewer: ```suggestion
The patterns described here are __not rules__, but should be taken as guidelines to when writing code in Rust.
```

Reviewer 2: ```suggestion
The patterns described here are __not rules__, but should be taken as guidelines when writing code in Rust.
```

Author: ```suggestion
The patterns described here are __not rules__, but should be taken as guidelines for writing idiomatic code in Rust.
```

?

Reviewer: That sounds much better to me, didn't even notice the extra `to` there, thanks pickfire!

Author: So pickfire's version it should be then, I assume?

> We are collecting these information so people can form a balanced view on certain aspects.

Reviewer: ```suggestion
We are collecting Rust patterns in this book so people can learn Rust idioms and use them in their own code.
```

Author: I miss the balanced view. The point was more to give information to the people so they can be informed about how things could be done and in the end they can decide with more knowledge about how they use things in their code. Hence we're trying to often give advantages and disadvantages to patterns, idioms etc.

Do you have a recommendation how to keep the mindset of "informing for a balanced view" in that sentence but make it better? :-P

Reviewer: ```suggestion
We are collecting Rust patterns in this book so people can learn the tradeoffs between Rust idioms and use them properly in their own code.
```

Maybe something like this?
I'm fine with the wording being changed around a bit on this, I made the changes to add `Rust patterns` and `Rust idioms` being explicit here because it's more direct to the reader.

> If you want to be part of that effort this is how you can participate:

Reviewer: ```suggestion
If you want to be part of this effort here are some ways you can participate:
```

## Discussion board

If you have a question or an idea regarding certain content but you want to have feedback of fellow community members 
and you think it may not be appropriate to file an issue open a discussion in our [discussion board](https://github.com/rust-unofficial/patterns/discussions).

> 

Reviewer 2: Why is the extra spaces necessary?

Author: Two new lines between a headline and the section before I thought

Reviewer: This should probably be enforced by a markdown formatter/linter.

Author: Yeah, we're working on it ;-)
